### PR TITLE
Entity and Combat

### PR DESCRIPTION
**Pull request by johne5s**
## All Submissions:

* [ ] I have read the LICENSE and agree to the terms for contributing Work.
* [ ] All 3rd party assets have been added to the `AssetManifest.list` and `.gitignore`
* [ ] All code submitted follows the [Coding Conventions](https://github.com/boppygames/community-game/wiki/coding-conventions)
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

## New Feature

1. [ ] All new assets (including texture, scripts, etc) are within a `Features/<feature>` directory
2. [ ] Any new art assets match the level of detail expected in the game (assets should be low-poly)
3. [ ] Screenshots added if necessary

### Description of feature

Include a short description here of the feature or core change that you made

### Screenshots

Include a couple of screenshots or gifs here to show off the new feature. This is optional for features that don't have a visual component to them.

## Bug Fixes

1. [ ] Describe the issue that existed before this PR
2. [ ] The bug fix has been tested 
3. [ ] Optional: Screenshots for before/after the bug fix

### CODE REVIEW

**Review by jdetter — CHANGES_REQUESTED** [review]
This is great so far, I have a few feedback items that I think can improve the code. Let me know what you think, and thanks for the contributions :)

File: Assets/Features/Johne5/EnemyController.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// Copyright 2021 Boppy Games, LLC
```
[the remark this answers is not in the document]
Comment (johne5s) [reply Assets/Features/Johne5/EnemyController.cs:1]: everything in the johne5 folder should be removed.  I moved all the files to Assets/Features/Entity/

File: Assets/Features/Combat/Scripts/EntityHealthController.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  [SerializeField]
+  int startingHealth;
+
+  [SyncVar] int health; //- the entity's current health value, should never be negative.
+
+  bool respawning = false;  // used to prevent multiple coroutines from running.
+
+  // Reduces the entity's health by the given damage value, sends `OnDeath` message if dead.
+  public void TakeDamage(int damage)
+  {
```
[the remark this answers is not in the document]
Comment (johne5s) [reply Assets/Features/Combat/Scripts/EntityHealthController.cs:32]: agree

File: Assets/Features/Combat/Scripts/EntityHealthController.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      {
+        StartCoroutine(RespawnCoroutine(5)); //respawn after 5 sec
+      }
+    }
+  }
+
+// When respawn is called, the health is set back to startingHealth and OnRespawn message is sent.
+  public void Respawn()
+  {
+    health = startingHealth;
```
[the remark this answers is not in the document]
Comment (jdetter) [reply Assets/Features/Combat/Scripts/EntityHealthController.cs:51]: Small suggestion: I think we should do a check at the top of this function
```C#
if (health > 0) return;
```

**Review by jdetter — CHANGES_REQUESTED** [review]
One more small suggestion, other than that I'm ready to merge this :+1: 

Thanks @johne5s!

**Review by jdetter — APPROVED** [review]
My suggestions here were optional, lets merge this :)

**Comment by jdetter** [comment]
Thanks for the contribution @johne5s!

**Pull request merged**